module/VuFind/src/VuFind/Controller/CombinedController.php

```diff
@@ -170,7 +171,21 @@ public function resultsAction()
         $initialType = $this->params()->fromQuery('type');
         foreach ($combinedOptions->getTabConfig() as $current => $settings) {
             [$searchClassId] = explode(':', $current);
-            $currentOptions = $optionsManager->get($searchClassId);
+            try {
+                $currentOptions = $optionsManager->get($searchClassId);
+            } catch (\Exception $e) {
+                // Prevent errors from any of the combined search results
+                // from raising up to the user interface and instead just skip them
+                $msg = "Failed get combined options for {$searchClassId}. {$e->getMessage()} ";
+                $this->log('err', [
+                    1 => $msg,
+                    2 => $msg,
+                    3 => $msg,
+                    4 => $msg . (string)$e,
+                    5 => $msg . (string)$e,
+                ]);
```

Comment:
How would you feel about this minor adjustment to reduce redundant messages in the messsage?

```suggestion
                $baseMsg = "Failed get combined options for {$searchClassId}.";
                $shortDetails = $e->getMessage();
                $fullDetails = (string)$e;
                $this->log('err', [
                    1 => "$baseMsg $shortDetails",
                    2 => "$baseMsg $shortDetails",
                    3 => "$baseMsg $shortDetails",
                    4 => "$baseMsg $fullDetails",
                    5 => "$baseMsg $fullDetails",
                ]);
```

Not sure if it also makes sense to move this logic into a trait so it's easier to reuse.

Final thought: `$this->log('err', ...)` is probably not the best way to log the message. It would be better to either use the `error()` method, or else to use a constant in place of `err`. Otherwise, there may be more work needed when it comes time to merge the monolog migration (though I think some work is going to be unavoidable in any scenario).

Reply:
I applied those changes but did need to pass in `$prependClass = false` since the `log()` function in the `LoggerAwareTrait` will make an assumption about the `$message` being a string if `$prependClass` is `true`.

It probably makes sense to update the `log()` function, but I wasn't sure if we wanted to get into touching the logging trait too much with the mololog migration happening at the same time in a separate PR. But maybe it would be something like:

```php
if ($this->logger) {
    if ($prependClass && is_string($message)) {
        $message = get_class($this) . ': ' . $message;
    }
    elseif ($prependClass && is_array($message)) {
        foreach ($message as &$value) {
            $value = get_class($this) . ': ' . $value;
        }
    }
    $this->logger->$level($message, $context);
}
```

I like the idea of making the log level logic a little more re-usable so it's less repetitive. But maybe it's worth doing that after the monolog PR is merged as a separate PR?

The logic could possibly even still just be in the `LoggerAwareTrait` `log()` function to have it parse the `$message` parameter and if it is an array with certain keys then it will convert it to the array with the keys `1`-`5` before calling `$this->logger->$level()`. We'd want it to be flexible enough to be able to specify when to switch over the verbosity level too since it might not always make sense for it to be levels 4 and 5 too. I started sketching out code for this and then it started getting into 40+ lines of code and I stopped myself. Likely something we will want to do in a separate PR!

Reply:
Yes, I agree that it's fine to leave this as-is for this PR, and we can explore deeper refactoring and improvement after the monolog migration is done. If you want to open a JIRA ticket to track your ideas, feel free, but if you don't think that's worth the effort, I can also just trust you to remember to circle back to this when the time is right!

There are still one or two outstanding issues in the monolog PR, but I have a good feeling that we'll get it finished sometime next week.

Reply:
I created the ticket just to get the idea out there because I think we'll need some opinions on how it be implemented and how flexible it should be. I put details in the ticket and the discussion on that can continue there.

https://openlibraryfoundation.atlassian.net/browse/VUFIND-1793

I assigned it to myself to plan on working on it once we get the details worked out on it (and monolog is merged).